DOCS-8064 ASM Exploit Prevention (#23704)
DataDog/documentation: content/en/security/application_security/threats/exploit-prevention.md

   a. Clone the policy. For example, **Managed - Block attack tools**.
    
    {{< img src="security/application_security/threats/clone-policy.png" alt="Clone the policy" width="100%" >}}
   b. Add a policy name and description.
        
    {{< img src="security/application_security/threats/clone-policy-modal.png" alt="Clone the policy" width="100%" >}}
   c. Click on the policy you created and select the **Local File Inclusion** ruleset and enable blocking for the **Local File Inclusion exploit** rule.
   
    {{< img src="security/application_security/threats/local-file-inclusion.png" alt="Local File Inclusion" width="100%" >}}
   d. Similarly, select the **Server-side Request Forgery** ruleset and enable blocking for the **Server-side request rorgery** exploit rule.
   
    {{< img src="security/application_security/threats/server-side-request-forgery.png" alt="Server-side Request Forgery exploit" width="100%" >}}
3. If you have applied a custom policy for your services, you can skip steps 2.a and 2.b for cloning a policy and directly set the Exploit Prevention rules in **blocking** mode (steps 2.c and 2.d).

## Reviewing exploit attempts in ASM

> Once you have enabled Exploit Prevention, if ASM detects an exploit attempt, it proceeds to block that request. In addition, Exploit Prevention detections are always accompanied by stack traces, which provide full visibility of where the vulnerability lies in the customer's code, ensuring a clear path to remediation.

Reviewer: ```suggestion
After you have enabled Exploit Prevention, if ASM detects an exploit attempt, it proceeds to block that request. Exploit Prevention detections are always accompanied by stack traces, which provide full visibility of where the vulnerability lies in your code, ensuring a clear path to remediation.
```

{{< img src="security/application_security/threats/exploit-prevention-detection.png" alt="Exploit Prevention detection" width="100%" >}}

> In addition, ASM also generates a signal correlating all the blocked traces and isolating the attacker IP addresses that are targeting your service(s). You can take a quick action to block all attacking IPs. 

Reviewer: ```suggestion
In addition, ASM also generates a signal correlating all the blocked traces and isolating the attacker IP addresses that are targeting your service(s). You can take action by blocking all attacking IPs. 
```

ASM Exploit Prevention inspects all file access attempts to determine if the path has been injected and whether a restricted file is accessed.

### Example 3: SQL injection

An attacker injects malicious SQL code into a query, potentially gaining unauthorized access to the database, manipulating data, or executing administrative operations.

ASM Exploit Prevention intercepts all SQL queries to determine if a user parameter has been injected and whether the injection alters the original purpose and structure of the SQL query.

## Prerequisites

- Ensure that your applications are instrumented with the Datadog tracer.
- ASM Threat Management must be enabled. See [Enabling ASM][1].
- Ensure Remote Configuration is enabled to push rule updates and In-App WAF policies. See [Enabling Remote Configuration][2].

### Library Compatibility

| Exploit Type                       | .NET             | Python           | Go               | Java, PHP, Node.js |
|------------------------------------|------------------|------------------|------------------|------------------------|

> | Server-side Request Forgery (SSRF) | v2.53.0          | v2.9.0           | v1.65.0          | Avail. in Q3' 24       |
> | Local File Inclusion (LFI)         | v2.53.0          | v2.9.0           | Avail. in Q3' 24 | Avail. in Q3' 24       |
> | SQL Injection (SQLi)               | Avail. in Q3 '24 | Avail. in Q3 '24 | Avail. in Q3 '24 | Avail. in Q3 '24       |

Reviewer: ```suggestion
| Server-side Request Forgery (SSRF) | v2.53.0          | v2.9.0           | v1.65.0          |        |
| Local File Inclusion (LFI)         | v2.53.0          | v2.9.0           |  |        |
```
